controllers/clusterwidenetworkpolicy_controller.go

```diff
@@ -181,3 +194,84 @@ func (r *ClusterwideNetworkPolicyReconciler) getReconciliationTicker(scheduleCha
 		}
 	}
 }
+
+func (r *ClusterwideNetworkPolicyReconciler) allowedCWNPsOrDelete(ctx context.Context, cwnps []firewallv1.ClusterwideNetworkPolicy, accessType firewallv2.NetworkAccessType, allowedNetworks firewallv2.AllowedNetworks) ([]firewallv1.ClusterwideNetworkPolicy, error) {
+	// FIXME refactor to func and add test, remove illegal rules from further processing
+	// report as event in case rule is not allowed
+
+	// what to do, if accesstype is baseline but there are allowedNetworks given?
+	if accessType != firewallv2.NetworkAccessForbidden {
+		return cwnps, nil
+	}
+
+	validCWNPs := make([]firewallv1.ClusterwideNetworkPolicy, 0, len(cwnps))
+	forbiddenCWNPs := make([]firewallv1.ClusterwideNetworkPolicy, 0)
+
+	egressSet, err := helper.BuildNetworksIPSet(allowedNetworks.Egress)
+	if err != nil {
+		return nil, err
+	}
+
+	ingressSet, err := helper.BuildNetworksIPSet(allowedNetworks.Ingress)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cwnp := range cwnps {
+		cwnp := cwnp
+		oke, err := r.validateCWNPEgressTargetPrefix(cwnp, egressSet)
+		if err != nil {
+			return nil, err
+		}
+		oki, err := r.validateCWNPIngressTargetPrefix(cwnp, ingressSet)
+		if err != nil {
+			return nil, err
+		}
+		// the CWNP is ok if both ingress/egress match with the allowed networks
+		ok := oki && oke
+
+		if !ok {
+			forbiddenCWNPs = append(forbiddenCWNPs, cwnp)
+		} else {
+			validCWNPs = append(validCWNPs, cwnp)
+		}
+	}
+	if len(cwnps) != len(validCWNPs) {
+		var errs []error
+		for _, cwnp := range forbiddenCWNPs {
+			cwnp := cwnp
+			err := r.ShootClient.Delete(ctx, &cwnp)
```

Comment:
keep the invalid cwnp, but update the cwnp.status with Error and Message which explains why.

The CWNP Printer should be extended with a Status Field that the user can easily see what happens

Reply:
This should be added to type ClusterwideNetworkPolicy struct below all other kubebuilder annotations
// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase"

Reply:
Would be great to add the status `Deployed` for all CWNPs as well